litellm/llms/watsonx/common_utils.py

```diff
@@ -279,13 +279,13 @@ def get_watsonx_credentials(
 
     def _prepare_payload(self, model: str, api_params: WatsonXAPIParams) -> dict:
         payload: dict = {}
+        payload["space_id"] = api_params["space_id"]
         if model.startswith("deployment/"):
             if api_params["space_id"] is None:
                 raise WatsonXAIError(
                     status_code=401,
                     message="Error: space_id is required for models called using the 'deployment/' endpoint. Pass in the space_id as a parameter or set it in the WX_SPACE_ID environment variable.",
                 )
-            payload["space_id"] = api_params["space_id"]
             return payload
         payload["model_id"] = model
```

Comment:
Unrelated to this issue, but this is fishy. I don't have a custom deployment to test this, but shouldn't this also get set before the return above?